allauth/socialaccount/providers/apple/provider.py

```diff
@@ -12,7 +12,19 @@ def extract_uid(self, data):
         return str(data['sub'])
 
     def extract_common_fields(self, data):
-        return dict(email=data.get('email'))
+        data = {
+            "email": data.get("email")
+        }
+
+        user_scope_data = data["user_scope_data"]
+        # Apple provide user name in nested json
+        # So check whether there is name
+        name = user_scope_data.get("name")
+        if name:
+            data["first_name"] = name.get("firstName", "")
+            data["last_name"] = name.get("lastName", "")
```

Comment:
@safwanrahman: What's the benefit of `user_scope_data` here? Can we not just use the existence of `firstName`/`lastName`? Seems like we'd be able to avoid the extra variable being parsed around in that case?

Reply:
@benhowes The problem is, apple is providing the user data in `user` to the callback and unlike other oath provider, it is not providing it inside the identity data. So we need to keep the data separate in other variable so that it does not get mixed with the identity data. [here](https://github.com/0leg5ergeev/django-allauth/pull/7/files#diff-579fe88cf513d91b95a7f0686be9bd05R143-R146) You can see that we are getting the value from `user` parameter which can be manipulated by user and the user can pass email as well. So I choose to keep it in separate variable.

Reply:
Yep, sorry got myself mixed up with that